torch/csrc/distributed/rpc/message.cpp

```diff
@@ -120,6 +130,48 @@ void Message::setId(int64_t id) {
   id_ = id;
 }
 
+at::IValue Message::toIValueTuple() const {
+  // Message has a std::vector<char> payload_ that is represented via a
+  // string, list of tensors, integer message type, and int64 id.
+  std::string payload(payload_.begin(), payload_.end());
+  c10::impl::GenericList messageTensors(at::TensorType::get());
+  messageTensors.reserve(tensors_.size());
+  for (const auto& tensor : tensors_) {
+    messageTensors.emplace_back(tensor);
+  }
+  return at::IValue(
+      c10::ivalue::Tuple::create(payload, messageTensors, type_, id_));
+}
+
+/* static */ Message Message::fromIValueTuple(const at::IValue& messageTuple) {
```

Comment:
Not sure if this is better as a static method or a constructor that takes in `at::IValue`.